### PR TITLE
MISUV-3592 added mfa credits to payment history page

### PR DESCRIPTION
**Pull request by joesheridan**

### CODE REVIEW

**Review by avelx — CHANGES_REQUESTED** [review]
minor changes as per comments

File: app/audit/models/PaymentHistoryResponseAuditModel.scala [hunk 1]
```diff
@@ -18,39 +18,43 @@ package audit.models
...
+    ("paymentDate", payment.date) ++
+    ("amount", payment.amount)
 
-
-  private def paymentHistoryDescriptionType(payment: Payment): JsObject = {
+  private def paymentHistoryMapper(payment: Payment): Option[JsObject] = {
     if (R7bTxmEvents) {
-      if (payment.credit.isDefined && CutOverCreditsEnabled) {
-        Json.obj("description" -> "Payment from an earlier tax year")
+      if (payment.credit.isDefined) {
```
[the remark this answers is not in the document]
Comment (joesheridan) [reply app/audit/models/PaymentHistoryResponseAuditModel.scala:40]: yep ok nice - will refactor

File: app/views/PaymentHistory.scala.html [hunk 2]
```diff
@@ -54,6 +54,14 @@
     }
 }
 
+@getMFACreditsLink(payment: Payment, isAgent: Boolean) = @{
+    if(isAgent) {
+        "/report-quarterly/income-and-expenses/view/agents/credits-from-hmrc/" + LocalDate.parse(payment.date.get).getYear
```
[the remark this answers is not in the document]
Comment (joesheridan) [reply app/views/PaymentHistory.scala.html:59]: yes, should be guarded - Ill ammend

File: conf/messages [hunk 3]
```diff
@@ -728,6 +728,7 @@ paymentHistory.textLink1                                        = Self Assessmen
 paymentHistory.button                                           = {0} payments
 paymentHistory.paymentToHmrc                                    = Payment made to HMRC
 paymentHistory.paymentFromEarlierYear                           = Payment from an earlier tax year
+paymentHistory.mfaCredit                                        = Credit from HMRC adjustment
```
Comment (avelx) [inline conf/messages:731]: pls add Welsh translation for this text resource

**Review by avelx — APPROVED** [review]

**Pull request merged**